core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -124,10 +124,16 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     (blockId, getFile(blockId))
   }
 
+  /**
+   * Create local directories for storing block data. These directories are
+   * located inside configured local directories and won't
+   * be deleted on JVM exit when using the external shuffle service.
```

Comment:
Can you please read [my comment](https://github.com/apache/spark/pull/4984#issuecomment-117351436) again? I tried to explain to the best of my ability. If that's not clear, I'm happy to try again, but please ask more specific questions.